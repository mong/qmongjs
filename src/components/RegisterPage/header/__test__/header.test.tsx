import React from "react";
import { BrowserRouter, Route } from "react-router-dom";
import { QueryParamProvider } from "use-query-params";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { Header } from "../";
import { RegisterNames } from "../../.";

const registerInfo: RegisterNames[] = [
  {
    id: 1,
    rname: "hjerneslag",
    full_name: "Norsk hjerneslagregister",
    caregiver_data: 1,
    resident_data: 1,
    dg_data: 1,
  },
  {
    id: 2,
    rname: "norgast",
    full_name: "Norsk register for gastrokirurgi",
    caregiver_data: 1,
    resident_data: 1,
    dg_data: 1,
  },
  {
    id: 3,
    rname: "barnekreft",
    full_name: "Nasjonalt kvalitetsregister for barnekreft",
    caregiver_data: 1,
    resident_data: 1,
    dg_data: 1,
  },
];
it("register page header renders correctly", () => {
  const registerNames = registerInfo;
  const tabNames = [
    { value: "sykehus", label: "Sykehus" },
    { value: "opptaksomraade", label: "Opptaksomr√•de" },
    { value: "datakvalitet", label: "Datakvalitit" },
  ];
  const dataFrom = "Norsk hjerneslagregister";

  const { container } = render(
    <BrowserRouter>
      <QueryParamProvider ReactRouterRoute={Route}>
        <Header
          activeTab={"sykehus"}
          registerNames={registerNames}
          tabNames={tabNames}
          dataFrom={dataFrom}
        />
      </QueryParamProvider>
    </BrowserRouter>
  );

  expect(screen.getByText(tabNames[0].label)).toBeInTheDocument();
  expect(screen.getByText(dataFrom)).toBeInTheDocument();
  expect(screen.getByText(registerNames[0].full_name)).toBeInTheDocument();
  expect(container).toMatchSnapshot();

  userEvent.click(screen.getByText(tabNames[1].label));
  expect(container).toMatchSnapshot();
});
