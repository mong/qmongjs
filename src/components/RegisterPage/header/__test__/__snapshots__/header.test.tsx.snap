// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register page header renders correctly 1`] = `
<div>
  <div
    class="headerOuterWrapper"
  >
    <div
      class="headerInnerWrapper"
    >
      <div
        class="topHeader"
      >
        <div
          class="topHeaderLeft"
        >
          <div
            class="nsmlogoWrapper"
          >
            <a
              href="https://www.kvalitetsregistre.no/"
            >
              <img
                alt="kvalitetsregistre.no"
                class="nsmlogo"
                src="nsmlogo.svg"
              />
            </a>
          </div>
          <div
            class="headerText"
          >
            <h3>
              Resultater fra 
              Norsk hjerneslagregister
            </h3>
          </div>
        </div>
        <div
          class="topHeaderRight"
        >
          <div
            style="position: sticky;"
          >
            <button
              class="registerBtn"
            >
              Velg register
            </button>
          </div>
          <div
            class="linkWrapper"
            style="display: none;"
          >
            <div
              class="searchInputWrapper"
            >
              <input
                class="searchInput"
                placeholder="Søk etter register"
                type="text"
                value=""
              />
            </div>
            <ul>
              <li>
                <a
                  href="/alle/sykehus"
                >
                  <b>
                    Alle registre
                  </b>
                </a>
              </li>
              <li>
                <a
                  href="/hjerneslag/sykehus"
                >
                  Norsk hjerneslagregister
                </a>
              </li>
              <li>
                <a
                  href="/norgast/sykehus"
                >
                  Norsk register for gastrokirurgi
                </a>
              </li>
              <li>
                <a
                  href="/barnekreft/sykehus"
                >
                  Nasjonalt kvalitetsregister for barnekreft
                </a>
              </li>
            </ul>
            <button
              class="select_reg_close"
            >
              <i
                class="far fa-times-circle"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="tabs"
      >
        <div
          class="tabsWrapper"
        >
          <ul
            class="tabsUL"
          >
            <li
              class="tabsLI"
            >
              <a
                aria-selected="false"
                class="tabsLink"
                href="/alle/sykehus"
                role="tab"
                style="color: rgb(109, 167, 223); box-shadow: -7px 7px 10px -5px #ccc; background-color: white; border-bottom: 3px solid white; border-radius: 5px;"
              >
                Sykehus
              </a>
            </li>
            <li
              class="tabsLI"
            >
              <a
                aria-selected="false"
                class="tabsLink"
                href="/alle/opptaksomraade"
                role="tab"
              >
                Opptaksområde
              </a>
            </li>
            <li
              class="tabsLI"
            >
              <a
                aria-selected="false"
                class="tabsLink"
                href="/alle/datakvalitet"
                role="tab"
              >
                Datakvalitit
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`register page header renders correctly 2`] = `
<div>
  <div
    class="headerOuterWrapper"
  >
    <div
      class="headerInnerWrapper"
    >
      <div
        class="topHeader"
      >
        <div
          class="topHeaderLeft"
        >
          <div
            class="nsmlogoWrapper"
          >
            <a
              href="https://www.kvalitetsregistre.no/"
            >
              <img
                alt="kvalitetsregistre.no"
                class="nsmlogo"
                src="nsmlogo.svg"
              />
            </a>
          </div>
          <div
            class="headerText"
          >
            <h3>
              Resultater fra 
              Norsk hjerneslagregister
            </h3>
          </div>
        </div>
        <div
          class="topHeaderRight"
        >
          <div
            style="position: sticky;"
          >
            <button
              class="registerBtn"
            >
              Velg register
            </button>
          </div>
          <div
            class="linkWrapper"
            style="display: none;"
          >
            <div
              class="searchInputWrapper"
            >
              <input
                class="searchInput"
                placeholder="Søk etter register"
                type="text"
                value=""
              />
            </div>
            <ul>
              <li>
                <a
                  href="/alle/sykehus"
                >
                  <b>
                    Alle registre
                  </b>
                </a>
              </li>
              <li>
                <a
                  href="/hjerneslag/sykehus"
                >
                  Norsk hjerneslagregister
                </a>
              </li>
              <li>
                <a
                  href="/norgast/sykehus"
                >
                  Norsk register for gastrokirurgi
                </a>
              </li>
              <li>
                <a
                  href="/barnekreft/sykehus"
                >
                  Nasjonalt kvalitetsregister for barnekreft
                </a>
              </li>
            </ul>
            <button
              class="select_reg_close"
            >
              <i
                class="far fa-times-circle"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="tabs"
      >
        <div
          class="tabsWrapper"
        >
          <ul
            class="tabsUL"
          >
            <li
              class="tabsLI"
            >
              <a
                aria-selected="false"
                class="tabsLink"
                href="/alle/sykehus"
                role="tab"
                style="color: rgb(109, 167, 223); box-shadow: -7px 7px 10px -5px #ccc; background-color: white; border-bottom: 3px solid white; border-radius: 5px;"
              >
                Sykehus
              </a>
            </li>
            <li
              class="tabsLI"
            >
              <a
                aria-selected="false"
                class="tabsLink"
                href="/alle/opptaksomraade"
                role="tab"
              >
                Opptaksområde
              </a>
            </li>
            <li
              class="tabsLI"
            >
              <a
                aria-selected="false"
                class="tabsLink"
                href="/alle/datakvalitet"
                role="tab"
              >
                Datakvalitit
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
